File: app/src/main/java/com/example/macarus0/walkiewalkie/data/Walk.java
package com.example.macarus0.walkiewalkie.data;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Ignore;
import android.arch.persistence.room.PrimaryKey;

import java.util.List;

@Entity
public class Walk {

    @PrimaryKey(autoGenerate = true)
    long walkId;
    String walkDate;
    int walkDistance;
    String walkDuration;
    long walkStartTime;
    long walkEndTime;
    int walkDogsCount;
    @Ignore
    List<Dog> dogs;

    public long getWalkStartTime() {
        return walkStartTime;
    }

    public void setWalkStartTime(long walkStartTime) {
        this.walkStartTime = walkStartTime;
    }

    public long getWalkEndTime() {
        return walkEndTime;
    }

    public void setWalkEndTime(long walkEndTime) {
        this.walkEndTime = walkEndTime;
    }

    public String getWalkDuration() {
        return walkDuration;
    }

    public void setWalkDuration(String walkDuration) {
        this.walkDuration = walkDuration;
    }

    @Ignore
    public long getWalkId() {
        return walkId;
    }

    @Ignore
    public void setWalkId(long walkId) {
        this.walkId = walkId;
    }

    @Ignore
    public String getWalkDate() {
        return walkDate;
    }

    @Ignore
    public void setWalkDate(String walkDate) {
        this.walkDate = walkDate;
    }

    public int getWalkDogsCount() {
        return walkDogsCount;
    }

    public void setWalkDogsCount(int walkDogsCount) {
        this.walkDogsCount = walkDogsCount;
    }

    @Ignore
    public List<Dog> getDogs() {
        return dogs;
    }

    @Ignore
    public void setDogs(List<Dog> dogs) {
        this.dogs = dogs;
    }

}

File: app/src/main/java/com/example/macarus0/walkiewalkie/data/Walk.kt
package com.example.macarus0.walkiewalkie.data

import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey

@Entity
class Walk {

    @PrimaryKey(autoGenerate = true)
    var walkId: Long = 0

    var walkDate: String = ""
    var walkDistance: Float = 0.toFloat()
    var walkDuration: String = ""
    var walkPathUrl: String = ""
    var walkStartTime: Long = 0
    var walkEndTime: Long = 0
    var walkDogsCount: Int = 0
    var isDistanceTracked: Boolean = false
    @Ignore
    var dogs: List<Dog> = emptyList()

    fun setWalkPathLink(walkPathLink: String) {
        this.walkPathUrl = walkPathLink
    }

}
